src/modules/tests/test_data/test-data.reducer.ts

```diff
@@ -57,6 +58,13 @@ export function testDataReducer(
         },
       };
     case testDataActions.REMOVE_DRIVING_FAULT:
+      if (action.payload.newFaultCount === 0) {
+        const { [action.payload.competency] : removedDrivingFault, ...updatedDrivingFaults } = state.drivingFaults;
```

Comment:
this puzzled me for a while until I realised the `...` in this context makes it a rest operator, not a spread! 😆

Reply:
Aye very cool, the _"dynamically qualified key rest operator"_, rolls straight off the tongue. 😉